pkg/timeline/service/push.ts

```diff
@@ -16,8 +19,74 @@ export class PushTimelineService {
     private readonly noteVisibility: NoteVisibilityService,
     private readonly timelineNotesCacheRepository: TimelineNotesCacheRepository,
     private readonly fetchSubscribedListService: FetchSubscribedListService,
+    /**
+     * @description Limit of timeline cache
+     * @private
+     */
+    private readonly TIMELINE_CACHE_LIMIT = 300,
   ) {}
 
+  /**
+   * @description Check the number of cached timelines and delete old ones if the limit is reached
+   * @private
+   */
+  private async timelineLimitCheck<T extends 'home' | 'list'>(
+    timelineType: T,
+    timelineID: T extends 'home' ? AccountID : ListID,
+  ): Promise<Result.Result<Error, void>> {
+    if (timelineType === 'home') {
+      const timeline = await this.timelineNotesCacheRepository.getHomeTimeline(
+        timelineID as AccountID,
+      );
+      if (Result.isErr(timeline)) {
+        return timeline;
+      }
+      const unwrappedTimeline = Result.unwrap(timeline);
```

Comment:
```suggestion
      const timelineRes = await this.timelineNotesCacheRepository.getHomeTimeline(
        timelineID as AccountID,
      );
      if (Result.isErr(timelineRes)) {
        return timelineRes;
      }
      const timeline = Result.unwrap(timelineRes);
```
`Result` のほうを `timelineRes` にして, `unwrap` 後のほうを `timeline` にすることを推奨します.